src/AdminConsole/EventLog/EventDeletionBackgroundWorker.cs

```diff
@@ -1,42 +1,35 @@
 using Passwordless.AdminConsole.EventLog.Loggers;
+using Passwordless.Common.Background;
 
 namespace Passwordless.AdminConsole.EventLog;
 
-public class EventDeletionBackgroundWorker : BackgroundService
+public class EventDeletionBackgroundWorker : BasePeriodicBackgroundService
 {
-    private readonly ILogger<EventDeletionBackgroundWorker> _logger;
     private readonly IServiceProvider _serviceProvider;
 
     public EventDeletionBackgroundWorker(
-        ILogger<EventDeletionBackgroundWorker> logger,
-        IServiceProvider serviceProvider)
+        IServiceProvider serviceProvider,
+        TimeProvider timeProvider,
+        ILogger<EventDeletionBackgroundWorker> logger)
+        : base(new TimeOnly(0), TimeSpan.FromDays(1), timeProvider, logger)
```

Comment:
Nit: Might be nice to have some `static readonly` "constants" that were a little more reader friendly. 

Thinking maybe a static class somewhere that had 
```
public static class TimeConstants 
{
    public static readonly TimeOnly Midnight = new TimeOnly(0);
}
```

Then when we see it, we instantly know "oh that's midnight."

Reply:
I need to move more to make it worth it. There is a couple more to move.